### PR TITLE
v2: Port profiles and import fixes

### PR DESCRIPTION
**Pull request by t1m0thyj**
**What It Does**

Ports misc fixes from the main branch:
* https://github.com/zowe/zowe-cli/pull/2313
* https://github.com/zowe/zowe-cli/pull/2343

**How to Test**

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -63,6 +64,11 @@ module.exports = {
             "ignoreEnums": true,
             "ignoreReadonlyClassProperties": true
         }],
+        "@typescript-eslint/no-restricted-imports": ["error", {
+            "patterns": [{
+                "group": ["**/../lib", "**/../src"]
+            }]
+        }],
```
Comment (traeok) [inline .eslintrc.js:71]: This lint rule seems really useful - would be nice to adopt it for Zowe Explorer & ZE API 😋

File: packages/imperative/src/config/src/ProfileInfo.ts [hunk 2]
```diff
@@ -316,8 +316,13 @@ export class ProfileInfo {
...
                 }
 
+                const updateVaultOnly = options.setSecure && this.getTeamConfig().api.secure.secureFields().includes(toUpdate.argLoc.jsonLoc);
                 this.getTeamConfig().set(toUpdate.argLoc.jsonLoc, options.value, { secure: options.setSecure });
-                await this.getTeamConfig().save(false);
+                if (!updateVaultOnly) {
+                    await this.getTeamConfig().save(false);
+                } else {
+                    await this.getTeamConfig().api.secure.save(false);
+                }
```
Comment (traeok) [inline packages/imperative/src/config/src/ProfileInfo.ts:325]: This probably sounds picky, but what are your thoughts on inverting the logic for the `if/else` for clarity? 😋 

```suggestion
                if (updateVaultOnly) {
                    await this.getTeamConfig().api.secure.save(false);
                } else {
                    await this.getTeamConfig().save(false);
                }
```

**Review by traeok — APPROVED** [review]
Thanks for addressing the typos and refactoring the profiles logic 😋 Had a suggestion for updating an if/else block, but aside from that everything LGTM

File: packages/imperative/src/config/src/ProfileInfo.ts (hunk 2 above)
Comment (t1m0thyj) [reply packages/imperative/src/config/src/ProfileInfo.ts:325]: I like this idea, but am reluctant to make the change here without making the same change in the V3 branch 😋

Comment (traeok) [reply packages/imperative/src/config/src/ProfileInfo.ts:325]: Ah, fair point - must have missed this in the v3 PR. Disregard this then, thanks for the consideration 😁

**Review by awharn — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
Thanks for porting! 🙏

**Pull request merged**